### PR TITLE
Add a setting to disable reading compile_commands.json for higher performance experience

### PR DESCRIPTION
**Pull request by xiaoyun94**
### This changes performance

The following changes are proposed:

## The purpose of this change
- Sacrifice some features for a higher performance experience.

### CODE REVIEW

**Review by elahehrashedi — CHANGES_REQUESTED** [review]

File: package.nls.json [hunk 1]
```diff
@@ -146,5 +146,6 @@
     "cmake-tools.configuration.cmake.launchBehavior.description": "Controls what happens with the launch terminal when you launch a target.",
     "cmake-tools.configuration.cmake.launchBehavior.reuseTerminal.description": "The launch terminal instance is reused and the target will launch as soon as the terminal is idle.",
     "cmake-tools.configuration.cmake.launchBehavior.breakAndReuseTerminal.description": "The launch terminal instance is reused and a break command is sent to terminate any active foreground process before launching the target.",
-    "cmake-tools.configuration.cmake.launchBehavior.newTerminal.description": "A new terminal instance is created and the target is launched in it. Existing terminals are not automatically cleaned up."
+    "cmake-tools.configuration.cmake.launchBehavior.newTerminal.description": "A new terminal instance is created and the target is launched in it. Existing terminals are not automatically cleaned up.",
+    "cmake-tools.configuration.cmake.omitLoadCompileCommand.description" : "Omit loading compile_commands.json, sacrifice some features for a higher performance experience."
```
Comment (elahehrashedi) [inline package.nls.json:150]: Please modify to:
`"cmake-tools.configuration.cmake.skipLoadingCompileCommands.description" : "Skip loading compile commands database. This leads to limited feature availability."`

File: src/cmakeTools.ts [hunk 2]
```diff
@@ -1182,8 +1182,9 @@ export class CMakeTools implements api.CMakeToolsAPI {
                 log.debug(localize('cannot.copy.compile.commands', 'Cannot copy {1} because it does not exist at {0}', compdbPath, 'compile_commands.json'));
             }
         }
-
-        if (compdbPaths.length > 0) {
+        if (this.workspaceContext.config.skipLoadingCompileCommands) {
```
Comment (elahehrashedi) [inline src/cmakeTools.ts:1185]: do we still need to find `compdbPaths` if we are going to set `compilationDatabase` to `null`?
I think we can check for `skipLoadingCompileCommands` earlier, before finding all `compdbPaths`

**Review by elahehrashedi — CHANGES_REQUESTED** [review]

**Comment by bobbrow** [comment]
Sorry for the silence on this.  We will re-evaluate for 1.13 after 1.12 is published.

File: src/cmakeTools.ts (hunk 2 above)
Comment (bobbrow) [reply src/cmakeTools.ts:1185]: The user may still want the copyCompileCommands setting to work

**Review by bobbrow — APPROVED** [review]

**Pull request merged**